src/modules/facility-loan/facility-loan.service.create-loan.test.ts

```diff
@@ -0,0 +1,205 @@
+      });
+
+      it(`uses productTypeId as loanInstrumentCode if productTypeId is '260'`, async () => {
+        const newLoanWithProductTypeId260 = {
+          ...newLoanGbp,
+          productTypeId: ENUMS.PRODUCT_TYPE_IDS.EWCS,
+        };
+        const acbsRequestBodyToCreateFacilityLoanWithProductTypeId260 = {
+          ...acbsRequestBodyToCreateFacilityLoanGbp,
+          BundleMessageList: [
+            {
+              ...acbsRequestBodyToCreateFacilityLoanGbp.BundleMessageList[0],
+              LoanInstrumentCode: ENUMS.PRODUCT_TYPE_IDS.EWCS,
+              ProductType: {
+                ProductTypeCode: ENUMS.PRODUCT_TYPE_IDS.EWCS,
+              },
+            },
+          ],
+        };
+        await service.createLoanForFacility(facilityIdentifier, newLoanWithProductTypeId260);
+
+        expect(acbsBundleInformationServiceCreateBundleInformation).toHaveBeenCalledWith(acbsRequestBodyToCreateFacilityLoanWithProductTypeId260, idToken);
+      });
+
+      it(`uses productTypeId as loanInstrumentCode if productTypeId is '280'`, async () => {
+        const newLoanWithProductTypeId280 = {
+          ...newLoanGbp,
+          productTypeId: ENUMS.PRODUCT_TYPE_IDS.GEF_CASH,
+        };
+        const acbsRequestBodyToCreateFacilityLoanWithProductTypeId280 = {
+          ...acbsRequestBodyToCreateFacilityLoanGbp,
+          BundleMessageList: [
+            {
+              ...acbsRequestBodyToCreateFacilityLoanGbp.BundleMessageList[0],
+              LoanInstrumentCode: ENUMS.PRODUCT_TYPE_IDS.GEF_CASH,
+              ProductType: {
+                ProductTypeCode: ENUMS.PRODUCT_TYPE_IDS.GEF_CASH,
+              },
+            },
+          ],
+        };
+        await service.createLoanForFacility(facilityIdentifier, newLoanWithProductTypeId280);
+
+        expect(acbsBundleInformationServiceCreateBundleInformation).toHaveBeenCalledWith(acbsRequestBodyToCreateFacilityLoanWithProductTypeId280, idToken);
+      });
+
+      it(`sets loanInstrumentCode to '280' if productTypeId is '281'`, async () => {
+        const newLoanWithProductTypeId281 = {
+          ...newLoanGbp,
+          productTypeId: ENUMS.PRODUCT_TYPE_IDS.GEF_CONTINGENT,
+        };
+        const acbsRequestBodyToCreateFacilityLoanWithProductTypeId281 = {
+          ...acbsRequestBodyToCreateFacilityLoanGbp,
+          BundleMessageList: [
+            {
+              ...acbsRequestBodyToCreateFacilityLoanGbp.BundleMessageList[0],
+              LoanInstrumentCode: ENUMS.PRODUCT_TYPE_IDS.GEF_CASH,
+              ProductType: {
+                ProductTypeCode: ENUMS.PRODUCT_TYPE_IDS.GEF_CONTINGENT,
+              },
+            },
+          ],
+        };
+        await service.createLoanForFacility(facilityIdentifier, newLoanWithProductTypeId281);
+
+        expect(acbsBundleInformationServiceCreateBundleInformation).toHaveBeenCalledWith(acbsRequestBodyToCreateFacilityLoanWithProductTypeId281, idToken);
+      });
+    });
+
```

Comment:
Can we create a test case where the loan bundle already exists please.

Reply:
newLoanRequests are not sent with an identifier so there's no a way of knowing if it already exists. Bundle identifier is in the response only

Reply:
Apologies for not clarifying, I meant can we create another loan with same payload please.
It should return back with two loan records (different bundle ID).

Reply:
Discussed with Abhi and Luke, this endpoint doesn't make a GET loans request to ACBS. It will always return a single Bundle ID response so an additional test is not required